types/canonical.go

```diff
@@ -23,14 +23,13 @@ type CanonicalPartSetHeader struct {
 }
 
 type CanonicalProposal struct {
-	Type             SignedMsgType // type alias for byte
-	Height           int64         `binary:"fixed64"`
-	Round            int64         `binary:"fixed64"`
-	POLRound         int64         `binary:"fixed64"`
-	Timestamp        time.Time
-	BlockPartsHeader CanonicalPartSetHeader
-	POLBlockID       CanonicalBlockID
-	ChainID          string
+	Type      SignedMsgType // type alias for byte
+	Height    int64         `binary:"fixed64"`
+	Round     int64         `binary:"fixed64"`
+	BlockID   CanonicalBlockID
+	POLRound  int64 `binary:"fixed64"`
```

Comment:
As far as I understand, the `BlockID` is not of fixed size (?). For HSMs to be able to parse the proposals, it is important that we keep the fixed sized fields in front (Type, Height, Round, POLRound, and then, BlockID, Timestamp, ChainID).